keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -198,8 +199,10 @@ class ApplicationController(
     @PathVariable("application") application: String,
     @RequestBody veto: EnvironmentArtifactVeto
   ) {
-
-    applicationService.markAsVetoedIn(application, veto, true)
+    val succeeded = applicationService.markAsVetoedIn(user, application, veto, true)
+    if (!succeeded) {
+      throw InvalidVetoException(application, veto.targetEnvironment, veto.reference, veto.version)
+    }
```

Comment:
a 'mark as bad' could fail if that version is pinned. in this case, we throw an error with a good user message.

Reply:
is it possible to throw that exception within `pplicationService.markAsVetoedIn`? it'll make the controller cleaner.

Reply:
sure